File: python/ch01/submit_flow.py
# python/ch01/submit_flow.py
import sys


def pre_submit_checks(output, validator, score_fn, timer):
    assert validator(output), "illegal output"
    score = score_fn(output)
    assert timer.elapsed() < timer.limit, "TLE locally"
    print("score:", score, file=sys.stderr)
    return output

File: go/ch01/submit_flow.go
// go/ch01/submit_flow.go
package main

import "fmt"

type State struct {
	Score int
	Data  []int
}

func bestSoFar(current, best State) State {
	if current.Score > best.Score {
		return current
	}
	return best
}

func main() {
	best := State{Score: -1}
	best = bestSoFar(State{Score: 100, Data: []int{1, 2}}, best)
	best = bestSoFar(State{Score: 250, Data: []int{3, 4}}, best)
	fmt.Println(best)
}
